### PR TITLE
3.x base_array_macro.html.twig updated as per latest twig

### PR DESCRIPTION
**Pull request by gomcodoctor**
base_array_macro.html.twig updated as per latest twig

otherwise it show 
Twig Runtime Error: Impossible to invoke a method (“render_array”) on a string variable

### CODE REVIEW

**Comment by jlamur** [comment]
You're fixing two distinct bugs here.

The first is that you need to use `{% from _self import render_array %}` to display multidimentional `array`s correctly with SF3.

The second one is that you want the type `array` to be able to display a `string`.

The first one is obviously a bug, while the second one is not and IMO this behaviour should not be implemented. If you want to display a `string` then use the `string` type.

@greg0ire @OskarStark please give your opinion.

**Comment by OskarStark** [comment]
> The first one is obviously a bug, while the second one is not and IMO this behaviour should not be implemented. If you want to display a string then use the string type.

I agree 👍

**Comment by greg0ire** [comment]
Yeah please make one commit per functional change

**Comment by gomcodoctor** [comment]
I have removed second behavior, only bug removed now

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already pushed. Don't forget the `--force` option otherwise git will try to merge both things together.

File: Resources/views/CRUD/base_array_macro.html.twig [hunk 1]
```diff
@@ -9,13 +9,13 @@ file that was distributed with this source code.
...
 {% macro render_array(value, inline) %}
+    {% from _self import render_array %}
     {% for key, val in value %}
         {% if val is iterable %}
-            [{{ key }} => {{ _self.render_array(val, inline) }}]
+            [{{ key }} => {{ render_array(val, inline) }}]
         {%  else %}
             [{{ key }} => {{ val }}]
         {%  endif %}
-
```
Comment (greg0ire) [inline Resources/views/CRUD/base_array_macro.html.twig]: Why did you remove this line?

Comment (gomcodoctor) [reply Resources/views/CRUD/base_array_macro.html.twig]: this was added to support both array and string in single macro, my concept was in case there is mixed data ( array / string ) then macro could handle both easily, but as you recommended that it should be one commit per functional change so I did it, sorry If i did something wrong, I am not expert in using github

Comment (greg0ire) [reply Resources/views/CRUD/base_array_macro.html.twig]: Can you add the line back then?

**Comment by greg0ire** [comment]
Hey there!

It looks like you did not fill the pull request template properly. Maybe you
were confused, so let me help you go through this step by step.

# Explaining why you target this branch

> I am targeting this branch, because {reason}.

Here, you are supposed to replace `{reason}` with something like:

- "there is a BC-break" if you picked the master branch
- "there is no BC-break" if you picked the stable branch

# Searching and finding related issues

> Closes #{put_issue_number_here}

Please have a look at the list of issues and pull requests, and look for issues
that will be fixed, and pull requests that will become irrelevant should we
merge your PR. Each issue / pull request has a number shown right next to the
title. Kindly replace `{put_issue_number_here}` with whatever issue or PR is
relevant. Of course, you may specify several numbers.
On merge, [these issues and PRs will automatically get
closed](https://help.github.com/articles/closing-issues-using-keywords/).
If there is no issue or pull request that fits there, just remove the sentence
entirely.

# Filling the changelog

That one is really important to get right, because it will be used to craft a
release changelog such as [this
one](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.23.0).
The first thing you need to know about this is that if your pull request is
pedantic, meaning it does not fix a bug for the user, or adds a new feature, it
is not needed. This is typically the case for PRs that fix the build or add
some documentation.

3 important things:

1. Make sure to fill the changelog section with something directed at the end
user, that answers the question "What benefit do I get from upgrading?".
2. Make sure to remove empty sections.
3. Make sure to keep the markdown markup.

## Filling the TODO list

If you think your PR is ready, just remove it. If you have still some things
left to do, customize it.

## Filling the Subject

Last but not least, describe your PR as well as you can. Bear in mind that some
reviewers do not even use the library they are maintaining, so try to be
crystal clear.
